frame/evm/src/runner/stack.rs

```diff
@@ -1170,9 +1184,35 @@ where
 			code.len(),
 			address
 		);
+
 		Pallet::<T>::create_account(address, code, caller)
 	}
 
+	fn set_delegation(
+		&mut self,
+		authority: H160,
+		delegation: evm::delegation::Delegation,
+	) -> Result<(), ExitError> {
+		log::debug!(
+			target: "evm",
+			"Inserting delegation (23 bytes) at {:?}",
+			delegation.address()
+		);
+
```

Comment:
In my opinion doing something like this, should be enough:

```rs
  if !<AccountCodes<T>>::contains_key(address) {
  	let account_id = T::AddressMapping::into_account_id(address);
  	T::AccountProvider::create_account(&account_id);
  }
  
  // Update metadata.
  let meta = CodeMetadata::from_code(&code);
  <AccountCodesMetadata<T>>::insert(address, meta);
  
  <AccountCodes<T>>::insert(address, code);
  Ok(())
```

Reply:
Addressed in https://github.com/polkadot-evm/frontier/pull/1727/commits/532a550b87ed1717ba5b3ce3e6d0f38fd3883b03